ecmd-core/capi/ecmdDataBufferBase.C

```diff
@@ -898,6 +922,10 @@ uint32_t ecmdDataBufferBase::shiftLeft(uint32_t i_shiftNum, uint32_t i_offset) {
     ETRAC2("**** ERROR : ecmdDataBufferBase::shiftLeft: i_offset %d > NumBits (%d)", i_offset, iv_NumBits);
     RETURN_ERROR(ECMD_DBUF_BUFFER_OVERFLOW);
   }
+  if (i_offset < i_shiftNum) {
```

Comment:
what bad behavior is this preventing?

Reply:
if the offset is less then the shiftnum we end up with a very large number when we attempt clearBit and this operation takes a long time and can often produce a segfault.  The clearbit in this method is taking a i_offset-i_shiftNum as the first parameter.